applicationset/generators/duck_type.go

```diff
@@ -136,18 +137,49 @@ func (g *DuckTypeGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.A
 
 	// Override the duck type in the status of the resource
 	statusListKey := "clusters"
-
-	matchKey := cm.Data["matchKey"]
-
 	if cm.Data["statusListKey"] != "" {
 		statusListKey = cm.Data["statusListKey"]
 	}
+
+	matchKey := cm.Data["matchKey"]
 	if matchKey == "" {
 		log.WithField("matchKey", matchKey).Warning("matchKey not found in " + cm.Name)
 		return nil, nil
 	}
 
+	clusterDecisions := buildClusterDecisions(duckResources, statusListKey)
+	if len(clusterDecisions) == 0 {
+		log.Warningf("clusterDecisionResource status.%s missing", statusListKey)
+		return nil, nil
+	}
+
 	res := []map[string]any{}
+	for _, clusterDecision := range clusterDecisions {
+		cluster := findCluster(clustersFromArgoCD, clusterDecision, matchKey, statusListKey)
+		if cluster == nil {
+			continue
+		}
+
+		// generated instance of cluster params
+		params := map[string]any{}
+		params["name"] = cluster.Name
+		params["server"] = cluster.Server
+
+		for key, value := range clusterDecision.(map[string]any) {
+			params[key] = value.(string)
+		}
```

Comment:
why is this required? Shouldn't we try to range over the local clusters that we have received from util/db package?